ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7906,6 +7945,10 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
     if (!(destType == QBMetaData.DEST_DFS_FILE && qb.getIsQuery())
             && destinationTable != null && destinationTable.getStorageHandler() != null) {
       try {
+        if (!updating(dest) && !deleting(dest)) {
+          input = genConversionSelectOperatorByAddPartition(dest, qb, input, destinationTable.getDeserializer(),
+                  destinationTable, partSpec);
+        }
         input = genConversionSelectOperator(
                 dest, qb, input, tableDescriptor.getDeserializer(conf), dpCtx, null, destinationTable);
```

Comment:
Two conversion SelectOperators. Are they necessary?

Reply:
@kasakrisz My idea was to isolate the addition of partition column into the operator with other cases where manipulation of select operator is necessary. Do you think it should within the same?